Binary files /home/hugo/Downloads/camlzip-1.05/dllcamlzip.so and ./dllcamlzip.so differ
diff -urN /home/hugo/Downloads/camlzip-1.05/Makefile ./Makefile
--- /home/hugo/Downloads/camlzip-1.05/Makefile	2012-10-11 20:36:07.000000000 -0700
+++ ./Makefile	2013-02-18 10:43:25.569962654 -0800
@@ -25,7 +25,7 @@
 
 all: libcamlzip.a zip.cma
 
-allopt: libcamlzip.a zip.cmxa
+allopt: libcamlzip.a zip.cmxa zip.cmxs
 
 zip.cma: $(OBJS)
 	$(OCAMLMKLIB) -o zip -oc camlzip $(OBJS) \
@@ -35,6 +35,9 @@
 	$(OCAMLMKLIB) -o zip -oc camlzip $(OBJS:.cmo=.cmx) \
             -L$(ZLIB_LIBDIR) $(ZLIB_LIB)
 
+zip.cmxs: zip.cmxa
+	$(OCAMLOPT) -shared -linkall -I ./ -o $@ $^
+
 libcamlzip.a: $(C_OBJS)
 	$(OCAMLMKLIB) -oc camlzip $(C_OBJS) \
             -L$(ZLIB_LIBDIR) $(ZLIB_LIB)
@@ -66,10 +69,11 @@
         fi
 
 installopt:
-	cp zip.cmxa zip.a zip.cmx gzip.cmx $(INSTALLDIR)
+	cp zip.cmxa zip.cmxs zip.a zip.cmx gzip.cmx $(INSTALLDIR)
 
 install-findlib:
-	ocamlfind install zip META *.mli *.a *.cmi *.cma $(wildcard *.cmxa) $(wildcard *.so)
+	ocamlfind install zip META *.mli *.a *.cmi *.cma $(wildcard *.cmxa) $(wildcard *.cmxs) $(wildcard *.so)
+	echo 'directory="../zip"' > META && ocamlfind install camlzip META
 
 depend:
 	gcc -MM -I$(ZLIB_INCLUDE) *.c > .depend
diff -urN /home/hugo/Downloads/camlzip-1.05/META ./META
--- /home/hugo/Downloads/camlzip-1.05/META	2012-10-11 20:36:07.000000000 -0700
+++ ./META	2013-02-18 10:07:16.585991713 -0800
@@ -2,3 +2,4 @@
 requires="unix"
 archive(byte)="zip.cma"
 archive(native)="zip.cmxa"
+archive(native,plugin)="zip.cmxs"
